onnxruntime/python/tools/quantization/matmul_4bits_quantizer.py

```diff
@@ -712,14 +712,20 @@ def process(self):
         if self.algo_config.algorithm in ["HQQ", "DEFAULT"]:
             # use a stack to keep track of sub-graphs
             graph_stack = [self.model.graph()]
-            opset_import = self.model.opset_import()
-
-            has_ms_domain = False
-            for opset in opset_import:
-                if opset.domain == "com.microsoft":
-                    has_ms_domain = True
-            if not has_ms_domain:
-                opset_import.extend([onnx.helper.make_opsetid("com.microsoft", 1)])
+
+            # Update domain opset
+            if self.algo_config.quant_format == QuantFormat.QOperator:
+                self.model.set_opset_import("com.microsoft", 1)
+            else:
+                opset_import = self.model.opset_import()
+                for opset in opset_import:
+                    if opset.domain == "ai.onnx" and opset.version < 21:
```

Comment:
>i [](http://example.com/codeflow?start=20&length=1)

empty name, i.e., "" also represents ai.onnx standard domain. you also need to check it.